src/proto/generation.proto

```diff
@@ -347,13 +342,48 @@ message TransformResample {
   optional bool export_mask = 5;               // return inpaint mask
 }
 
+message PointCloudRenderParameters {
+  float radius = 1;            // The radius (in NDC units) of each disk to be rasterized 
+  uint32 points_per_pixel = 2;  // Number of points to keep track of per pixel
+}
+
+message MeshRenderParameters {
+  float max_mesh_edge = 1;  // Mesh edges longer than this value are discarded
+}
+
+message RenderParameters {
+  oneof render_parameters {
+    PointCloudRenderParameters pointcloud_parameters = 1;
+    MeshRenderParameters mesh_parameters = 2;
+  }
+}
+
+enum CameraType {
+  CAMERA_PERSPECTIVE = 0;   // Perspective camera
+  CAMERA_ORTHOGRAPHIC = 1;  // Orthographic camera. Well suited for isometric animations
+}
+
+message CameraParameters {
+  CameraType camera_type = 1;
+  float near_plane = 2;    // Nearest plane depth of a rendered frustum
+  float far_plane = 3;     // Farthest plane depth of a rendered frustum
+  optional float fov = 4;  // Camera field of view (in degrees). Must be set for CAMERA_PERSPECTIVE type.
+}
+
+message TransformCameraPose {
+  TransformMatrix world_to_view_matrix = 1;  // 4x4 transform matrix for the next frame
+  CameraParameters camera_parameters = 2;
+  RenderParameters image_render_parameters = 3;
+  RenderParameters mask_render_parameters = 4;
+  bool do_prefill = 5;   // Prefill masked areas with values matching the colors around the area
+}
+
 message TransformParameters {
   oneof transform {
-    TransformBlend blend = 1;
     TransformColorAdjust color_adjust = 2;
-    TransformColorMatch color_match = 3;
```

Comment:
1 and 3 have been removed, should we make them as _reserved_ to avoid future conflicts?

Reply:
The reserved syntax isn't valid inside `oneof` types unfortunately. There are no external clients we have to worry about using the field numbers since we have not deployed yet so using 1 and 3 in future wouldn't be a problem but by convention I think we should always be adding new fields as the next numeric value anyway which will be 7 here.